host/src/connection.rs

```diff
@@ -191,24 +172,19 @@ impl<'stack> Connection<'stack> {
         self.manager.alloc_tx()
     }
 
-    /// Wait for next connection event.
+    /// Wait for next connection event data.
     #[cfg(not(feature = "gatt"))]
-    pub async fn next(&self) -> ConnectionEvent {
+    pub(crate) async fn next(&self) -> ConnectionEventData {
         match self.manager.next(self.index).await {
-            ConnectionEventData::Disconnected { reason } => ConnectionEvent::Disconnected { reason },
+            ConnectionEventData::Disconnected { reason } => ConnectionEventData::Disconnected { reason },
             ConnectionEventData::Gatt { data } => unreachable!(),
         }
     }
 
-    /// Wait for next connection event.
+    /// Wait for next connection event data.
     #[cfg(feature = "gatt")]
-    pub async fn next(&self) -> ConnectionEvent<'stack> {
-        match self.manager.next(self.index).await {
-            ConnectionEventData::Disconnected { reason } => ConnectionEvent::Disconnected { reason },
-            ConnectionEventData::Gatt { data } => ConnectionEvent::Gatt {
-                data: crate::gatt::GattData::new(data, self.clone()),
-            },
-        }
+    pub(crate) async fn next(&self) -> ConnectionEventData {
+        self.manager.next(self.index).await
```

Comment:
I'd like to keep ConnectionEvent and ConnectionEventData as is, because:

1. There is a use of reading connection events for non-GATT applications to discover disconnections.
2. We currently support the use case of handling GATT data without relying on the AttributeServer, which would not be possible anymore with this change. Using the attribute server is a trade-off between convenience and memory usage.